src/types/closure.md

```diff
@@ -178,3 +293,258 @@ Because captures are often by reference, the following general rules arise:
+        * else:
+            * Add `move_xform(Place1, Mode1)` to C'
+* Minimization
+    * Until no rules apply:
+        * For each two places (P1, M1), (P2, M2) where P1 is a prefix of P2:
+            * Remove both places from the set
+            * Add (P1, max(M1, M2)) into the set
+* Helper functions:
+    * `unsafe_check(Place, Mode) -> (Place, Mode)`
+        * "Ensure unsafe accesses occur within the closure"
+        * If Place contains a deref of a raw pointer:
+            * Let Place1 = Place truncated just before the deref
+            * Return (Place1, Mode)
+        * If Mode is `ref *` and the place contains a field of a packed struct:
+            * Let Place1 = Place truncated just before the field
+            * Return (Place1, Mode)
+        * Else
+            * Return (Place, Mode)
+    * `move_xform(Place, Mode) -> (Place, Mode)` (For move closures)
+        * "Take ownership if data being accessed is owned by the variable used to access it (or if closure attempts to move data that it doesn't own)."
+        * "When taking ownership, only capture data found on the stack."
+        * "Otherwise, reborrow the reference."
+        * If Mode is `ref mut` and the place contains a deref of an `&mut`:
+            * Return (Place, Mode)
+        * Else if Mode is `ref *` and the place contains a deref of an `&`:
+            * Return (Place, Mode)
+        * Else if place contains a deref:
+            * Let Place1 = Place truncated just before the deref
+            * Return (Place1, ByValue)
+        * Else:
+            * Return (Place, ByValue)
+    * `ref_xform(Place, Mode) -> (Place, Mode)` (for ref closures)
+        * "If taking ownership of data, only move data from enclosing stack frame."
+        * Generate C' by mapping each (Mode, Place) in C
+            * If Mode is ByValue and place contains a deref:
+                * Let Place1 = Place truncated just before the deref
+                * Return (Place1, ByValue)
+            * Else:
+                * Return (Place, Mode)
+    * `ref_opt(Place, Mode) -> (Place, Mode)`
+        * "Optimization: borrow the ref, not data owned by ref."
+        * Disjoint capture over immutable reference doesn't add too much value because the fields can either be borrowed immutably or copied.
+        * Edge case: Field that is accessed via the referece lives longer than the reference.
+            * Resolution: Only consider the last Deref
+        * If Place is (Base, Projections), where Projections is a list of size N.
+            * For all `i, 0 <= i < N`, Projections[i] != Deref
+                * Return (Place, Mode)
+            * If `l, 0 <= l < N` is the last/rightmost Deref Projection i.e. for any `i, l < i < N` Projection[i] != Deref,
+              and `Place.type_before_projection(l) = ty::Ref(.., Mutability::Not)`
+                * Let Place1 = (Base, Projections[0..=l])
+                * Return (Place1, Ref)
+
+## Key examples
+
+### box-mut
```

Comment:
Let's add a comment here:

This test shows how a `move` closure can sometimes capture values by mutable reference, if they are reached via a `&mut` reference.